SAP/open-ux-tools
# Support for SAP UI5 adaptation projects.

## Description
Backlog item: #971

## Details
The PR contains 4 major parts:
* a small enhancement of the `@sap-ux/axios-extension` required to merge app descriptor variants
* a new module called `@sap-ux/adp-tooling` containing preview and generation functionality for SAP UI5 adaptation projects
* an enhancement of the `@sap-ux/preview-middleware` using the preview functionality of the new `@sap-ux/adp-tooling` module
* an enhancement of the `@sap-ux/create` module using the writer functionality of the new `@sap-ux/adp-tooling` module

## Testing it manually
The best and easiest way to test it, is to create a project with the enhanced `@sap-ux/create` module and then try the preview in the generated project.
1. Ensure all projects are built with `pnpm build`
2. Go to `packages/create`
3. Execute `node ./dist/index.js generate adaptation-project .tmp/mytest --id my.id --reference the.other.app --url http://your.systm.example --skip-install`
4. Go to `./tmp/mytest
5. Modify the `package.json` and the `ui5.yaml` to use the local `preview-middleware` and not the last release:
   1. remove the `devDependency` of `@sap-ux/preview-middleware`
   2. add a local reference to the ui5.yaml by adding at the end
```
---
specVersion: "3.0"
metadata:
    name: preview-middleware
kind: extension
type: server-middleware
middleware:
    path: ../../../preview-middleware/dist/ui5/middleware.js`;
```
6. run `npm i`
7. Run `npm start` or `npm run editor` 

You can of course also remove the parameters in step 3 and use the prompt to enter your test values.
